### ISSUE TITLE
Make sure all code insights features are in enterprise directories

### ISSUE THREAD

**Issue by Joelkw** [sourcegraph/sourcegraph#24201]
We should make sure things for enterprise (which is all of code insights, currently) are in enterprise folders. [More context here](https://docs.google.com/document/d/1qNDIDgF__RmupRDid5YUoMU6Tb01GXExkQbstlGfjo8/edit#)

**Comment by vovakulikov** [comment]
At the moment the whole Code Insights FE is located in OSS version `./client/web/src/insights/**/*`.
So to make code insights FE enterprise-based we need to move this folder to `/client/web/src/enterprise/insight` 

The only problem I see (it isn't a problem but a thing which we shouldn't miss) some of our OSS components use Insight components, like TreePage and HomePage. We need to make sure that these insight shared components are still placed in OSS `./client/web/src/insights/**/*`. Cause we can't have imports from enterprise to OSS version. 

Components like this: 
- `StaticInsightsViewGrid.tsx` and all components that are used by this component 
   - `ViewGrid.tsx`
   - `InsightViewContent.tsx`
   - `ViewInsightProviderResult` interface from `/insights/core/backend/types`
- `getCombinedViews` API fetcher also is needed for TreePage and Homepage insight loading 

cc @felixfbecker @unclejustin

**Comment by felixfbecker** [comment]
> We need to make sure that these insight shared components are still placed in OSS ./client/web/src/insights/**/*. Cause we can't have imports from the enterprise to the OSS version.

It doesn't necessarily mean that – the way that many areas of the app work that render additional content in enterprise is that configuration is passed in at the root about which sections should appear. For example, different nav items in the site admin section. One could imagine a new configuration `treePageSections` where the insights grid is an item added in enterprise.

If that was not done, you'd usually end up effectively having to keep everything in OSS (more than intended, because of transitive dependencies).

However, we also have the `registerViewProvider()` extension API, which we probably can/should keep intact for OSS. I think the right split would be to still support the grid and the "views" (including chart content) contributed by extensions on the TreePage, but move the dashboards pages, creation UI, and backend fetching into enterprise. Since we don't show backend insights on directory pages and the home page, we could safely do that without extra effort.

Basically in the end I'd propose we'd make it so that everything that has "insights" in its name is in enterprise, and everything that is really just "views" related or "grid" related is in OSS (there may also be some things that currently don't fit that naming scheme which should be renamed rather than moved).

**Comment by vovakulikov** [comment]
@felixfbecker thanks for your thoughts. I agree with the idea about splitting up visual components (I remember we had views directory at some point, but I deleted it) that are used by OSS pages and some code insight-specific components and pages under the enterprise. We also share some fetch code insights for the Tree and Search pages but I think it's fine to leave this at the OSS insight directory since it's related only to the OSS part and we don't use it for the dashboard page.

**Comment by felixfbecker** [comment]
> We also share some fetch code insights for the Tree and Search pages but I think it's fine to leave this at the OSS insight directory since it's related only to the OSS part

What do they "fetch" from the backend?

**Comment by vovakulikov** [comment]
@felixfbecker Tree page and Search page, they [use `getCombinedViews` to fetch insights]( https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/repo/tree/TreePage.tsx?L279) (both type, backend, and extension-based)

[getCombinedViews](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/core/backend/api/get-combined-views.ts?L15:1) uses [getBackendInsights](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/core/backend/api/get-backend-insights.ts?L13:17) handler to fetch BE insights via our gql API

I think that we should extract this method (`getCombinedViews`) from our Insight API context and use it as a helper function at the Tree and Search pages. And then we can move the Insight API context to the enterprise directory.

**Comment by vovakulikov** [comment]
@unclejustin Let me know what is the best day to merge such a PR with moving a lot of things to a different folder. As we discussed that I think we should sync before do that to avoid complex rebase conflicts. My suggestion is let finish this week with ongoing issues tickets and make a PR with these changes on Monday. I can do it before your working hours and then you can review it before starting another story/ticket/issue. Let me know if it doesn't work for you 🐱

**Comment by unclejustin** [comment]
@vovakulikov that sounds perfect.

**Comment by felixfbecker** [comment]
As part of this issue we should also fix https://github.com/sourcegraph/sourcegraph/issues/20178
